assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.js

```diff
@@ -0,0 +1,100 @@
+(function() {
+    'use strict';
+
+    /**
+    * @ngdoc function
+    * @name crowbarApp.controller:Upgrade7OpenStackServicesController
+    * @description
+    * # Upgrade7OpenStackServicesController
+    * This is the controller used on the Stop OpenStack Services page
+    */
+    angular.module('crowbarApp')
+        .controller('Upgrade7OpenStackServicesController', Upgrade7OpenStackServicesController);
+
+    Upgrade7OpenStackServicesController.$inject = ['$translate', 'openStackFactory'];
+    // @ngInject
+    function Upgrade7OpenStackServicesController($translate, openStackFactory) {
+        var vm = this;
+            
+
+        vm.openStackServices = {
+            completed: false,
+            valid: false,
+            stopServicesErrorMessage: false,
+            checks: {
+                services: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.services'
+                },
+                backup: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.backup'
+                }
+            },
+            runOpenStackServices: runOpenStackServices
+        };
+
+
+        /**
+         *  Validate OpenStackServices required for Cloud 7 Upgrade
+         */
+        function runOpenStackServices() {
+   
+            openStackFactory.stopOpenstackServices()
+                .then(
+                    //Success handler. Stop all OpenStackServices successfully:
+                    function(openStackServicesResponse) {
+
+                        vm.openStackServices.checks.services.status = openStackServicesResponse.data.services;
+                        
+                        if (true === vm.openStackServices.checks.services.status) {
+
+                            openStackFactory.createOpenstackBackup()
+                                .then(
+                                    //Success handler. Backup OpenStackServices successfully:
+                                    function(openStackServicesResponse) {
+                                        var ResponseData = openStackServicesResponse.data
+
+                                        vm.openStackServices.checks.backup.status = ResponseData.backup;
+                                        vm.openStackServices.valid = vm.openStackServices.checks.backup.status;
+
+                                    },
+                                    //Failure handler:
+                                    function(errorOpenStackServicesResponse) {
+
+                                        // Expose the error list to openStackServices object
+                                        vm.openStackServices.errors = errorOpenStackServicesResponse.data.errors;
+                                    }
+                                )
+                        } else { 
+                            vm.openStackServices.stopServicesErrorMessage = true;
```

Comment:
I think we can rely on the view to check the status of each check to decide wether to display an error or not. That way we can remove this attribute. What do you think?